rspirv/sr/context.rs

```diff
@@ -54,58 +58,88 @@ impl<T> PartialEq for Token<T> {
 impl<T> Eq for Token<T> {}
 
 impl<T> Token<T> {
-    pub(in crate::sr) fn new(index: Index) -> Self {
+    pub const DUMMY: Self = Token {
+        index: !0,
+        marker: PhantomData,
+    };
+
+    fn new(index: Index) -> Self {
         Token {
             index,
             marker: PhantomData,
         }
     }
 }
 
+
+/// Generic data that has an associated reference to the source SPIR-V.
+#[derive(Debug)]
+struct Data<T> {
+    value: T,
+    raw_index: Option<NonZeroU32>,
+}
+
 /// A structure holding some kind of SPIR-V entity (e.g., type, constant,
 /// instruction, etc.) that can be referenced.
 #[derive(Debug)]
 pub struct Storage<T> {
-    map: FastHashMap<Index, T>,
-    next_id: Index,
+    /// Values of this storage.
+    data: Vec<Data<T>>,
+    /// Reverse lookup table that associates SPIR-V references with tokens.
```

Comment:
Can we be consistent on terms used? It will make the code and comment more consistent and lower the barrier for understanding the code.

We should use `<id>`, which is used in the SPIR-V spec consistently, to mean the "references" in the raw binary and DR. We should use `Token` to "refer" instructions in SR. I think `Index` is just an implementation detail to SR that implies we are using a `Vec` internally to store all the instructions in SR. Public users should not be concerned about that.

Reply:
I agree about the consistency in terms. Just need to get used to them :)

As for `Index` - it is still private, and this `lookup` field is private.